app/controllers/comments_controller.rb

```diff
@@ -5,7 +5,8 @@ class CommentsController < ApplicationController
 
   def create
     @post = Post.find(params[:post_id])
```

Comment:
@membrq The `params` contents can be used to figure out if you're dealing with a topic comment or post comment:

```rb
def create
  # There's a `post_id` param, meaning this request is for creating
  # a comment for a post
  is_comment_for_post = params[:post_id].present?
  # Not needed but would also work:
  # is_comment_for_topic = params[:topic_id].present?
  if is_comment_for_post
    # Do post-specific work in here
  else
    # This isn't a comment for a post so it must be a comment for a topic.
    # Do topic-specific work in here
  end
  ...
```